src/Microsoft.TestPlatform.CrossPlatEngine/TestEngine.cs

```diff
@@ -224,31 +284,61 @@ public TestEngine() : this(TestRuntimeProviderManager.Instance, new ProcessHelpe
             Func<ProxyOperationManager> proxyCreator = () =>
             {
                 var hostManager = this.testHostProviderManager.GetTestHostManagerByRunConfiguration(testSessionCriteria.RunSettings);
-                hostManager?.Initialize(TestSessionMessageLogger.Instance, testSessionCriteria.RunSettings);
+                if (hostManager == null)
+                {
+                    throw new TestPlatformException(
+                        string.Format(
+                            CultureInfo.CurrentCulture,
+                            Resources.Resources.NoTestHostProviderFound));
+                }
 
+                hostManager.Initialize(TestSessionMessageLogger.Instance, testSessionCriteria.RunSettings);
                 if (testSessionCriteria.TestHostLauncher != null)
                 {
                     hostManager.SetCustomLauncher(testSessionCriteria.TestHostLauncher);
                 }
 
-                var requestSender = new TestRequestSender(requestData.ProtocolConfig, hostManager);
+                var requestSender = new TestRequestSender(requestData.ProtocolConfig, hostManager)
+                {
+                    CloseConnectionOnOperationComplete = false
+                };
 
+                // TODO (copoiena): For now we don't support data collection alongside test
```

Comment:
This shouldn't break any functionality because it is in `GetTestSessionManager` which is added by me in order to support test sessions (discovery/run behavior remains the same, they do support data collection). The support for this feature is not yet complete (hence the `Obsolete` attributes on the main entry points in `TranslationLayer`) and as such shouldn't be used until it is publicly announced because it's subject to changes. The API should be final only after TE successfully integrates our changes.